res/controllers/Denon Prime 4.midi.xml

```diff
@@ -0,0 +1,1735 @@
+<?xml version='1.0' encoding='utf-8'?>
+<MixxxControllerPreset schemaVersion="1" mixxxVersion="2.4.0+">
+    <info>
+        <name>Denon Prime 4 - Mixxx Mapping</name>
+        <author>Levi "Whanake" Williams</author>
+        <description>This is a Work-In-Progress mapping to allow the Denon Prime 4 to be used with Mixxx, ideally with the same (if not more) functionality that the Prime 4 provides as a standalone device.</description>
+        <wiki>https://github.com/mixxxdj/mixxx/wiki/Denon-Prime-4</wiki>
+        <forum>https://mixxx.discourse.group/t/denon-prime-4-mapping/22404</forum>
+    </info>
+    <controller id="denonprime4">
+        <scriptfiles>
+            <file filename="lodash.mixxx.js"/>
+            <file filename="midi-components-0.0.js"/>
+            <file filename="Denon-Prime-4-scripts.js" functionprefix="Prime4"/>
+        </scriptfiles>
+        <controls>
+            <!-- FX RACKS + MASTER SECTION-->
+            <control> <!-- Master Gain Knob -->
+              <group>[Master]</group>
+              <key>gain</key>
+              <description>MIDI Learned from 506 messages.</description>
+              <status>0xBF</status>
+              <midino>0x0F</midino>
+              <options>
+                <normal/>
+              </options>
+            </control>
+
+            <control> <!-- FX1 - Effect 1 Button -->
+			  <group>[EffectRack1_EffectUnit1_Effect1]</group>
+              <status>0x96</status>
+              <midino>0x06</midino>
+              <options>
+                <normal/>
+              </options>
+            </control>
```

Comment:
this doesn't do anything because the `key` is missing.

Reply:
That's intentional. I haven't begun work on the effects units yet, these are just here as placeholders